### PR TITLE
Feature/vta 379

### PR DESCRIPTION
**Pull request by wis-dvsa**

### CODE REVIEW

File: src/test/java/testresults/PostTestResultsProvisionalUpdateFirstTestTrl.java [hunk 1]
```diff
@@ -7,7 +7,9 @@
 import net.thucydides.core.annotations.Title;
 import net.thucydides.junit.annotations.TestData;
 import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Before;
 import org.junit.Ignore;
```
Comment (IuliaDVSA) [inline src/test/java/testresults/PostTestResultsProvisionalUpdateFirstTestTrl.java:11]: Is the ignore import needed>?

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java [hunk 2]
```diff
@@ -23,7 +23,6 @@
     @TestData
     public static Collection<Object[]> testData(){
         return Arrays.asList(new Object[][]{
-                {"warmup test", "warmup test", "95", 1, "pass", "ffv2"},
```
Comment (IuliaDVSA) [inline src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java:26]: Is this warm-up not needed?

File: src/test/java/trailerRegistration/PostTrailerRegistration.java [hunk 3]
```diff
@@ -496,8 +510,8 @@ public void testPostTestResultFailTrailerRegistrationCertificateWithoutTrn() {
 
         testResultsSteps.validateCertificateIsGenerated(testNumber,randomVin);
     }
-
-    @Title("CVSB-18921 - AC4 -  VTG30 NO TRN PRESENT - HGV Annual Test Fail - Without TRN")
+//this one
```
Comment (IuliaDVSA) [inline src/test/java/trailerRegistration/PostTrailerRegistration.java:513]: Please remove the comment or expand to explain following line

**Review by IuliaDVSA — COMMENTED** [review]
A few comments pending reply. We discussed about the 504, did not comment on that :)

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java (hunk 2 above)
Comment (wis-dvsa) [reply src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java:26]: not at the moment, the test passes without it

**Review by IuliaDVSA — APPROVED** [review]

File: src/main/java/data/GenericData.java [hunk 4]
```diff
@@ -456,4 +459,59 @@ public static JSONObject getRestrictions(String testCode) {
         }
         return actualRestrictions;
     }
```
Comment (cb-cs) [inline src/main/java/data/GenericData.java:461]: There is a fair bit of duplicate logic in these 2 methods, using a flag or additional parameter should allow you to amalgamate them into 1.

File: src/test/java/testresults/GetTestResults.java [hunk 5]
```diff
@@ -41,6 +42,18 @@
     private TestResultsGet vehicleSubmittedData = TestResultsData.buildTestResultsSubmittedDataWithCalculated().build();
     private TestResultsGet vehicleCancelledData = TestResultsData.buildTestResultsCancelleddDataWithCalculated().build();
 
+    private String test_results_roadworthiness_hgv_pass_7675_json;
```
Comment (cb-cs) [inline src/test/java/testresults/GetTestResults.java:45]: vehicleTechnicalRecordsSteps on line 38 is greyed out, is that required now?

File: src/main/java/data/GenericData.java [hunk 6]
```diff
@@ -456,4 +459,59 @@ public static JSONObject getRestrictions(String testCode) {
...
+        String body4 =  body3.replaceAll("testTypeStartTimestamp.*?\\,", "testTypeStartTimestamp\":\""+startTime+"\",");
+
+        String body5 =  body4.replaceAll("testTypeEndTimestamp.*?\\,", "testTypeEndTimestamp\":\""+endTime+"\",");
+
+        String finalBody = body5;
+
+        return finalBody;
+    }
+
+    public static String updateJson2(TestResultsSteps testResultsSteps, String jsonFileName, String $) {
```
Comment (cb-cs) [inline src/main/java/data/GenericData.java:489]: testResultSteps passed into this method is no longer being used, is it still required?

File: src/test/java/testresults/GetTestResultsApiSpecs.java [hunk 7]
```diff
@@ -25,12 +26,31 @@
 
     private TestResults.Builder vehicleSubmittedData = TestResultsData.buildTestResultsSubmittedData();
```
Comment (cb-cs) [inline src/test/java/testresults/GetTestResultsApiSpecs.java:27]: vehicleSubmittedData on line 27 is greyed out, is that required now?

File: src/test/java/testresults/GetTestResultsBetweenDate.java [hunk 8]
```diff
@@ -30,12 +31,20 @@
     private TestResultsGet vehicleSubmittedDataOld = TestResultsData.buildTestResultsSubmittedDataWithCalculated().build();
     private TestResultsGet vehicleCancelledData = TestResultsData.buildTestResultsCancelleddDataWithCalculated().build();
 
+    private String test_results_roadworthiness_hgv_pass_7675_json;
```
Comment (cb-cs) [inline src/test/java/testresults/GetTestResultsBetweenDate.java:34]: vehicleSubmittedDataOld on line 31 is greyed out, is that required now?

File: src/test/java/testresults/GetTestResultsToDate.java [hunk 9]
```diff
@@ -8,6 +8,7 @@
 import net.thucydides.core.annotations.Steps;
 import net.thucydides.core.annotations.Title;
 import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Before;
 import org.junit.Ignore;
```
Comment (cb-cs) [inline src/test/java/testresults/GetTestResultsToDate.java:12]: Unused umport

File: src/test/java/testresults/GetTestResultsVinSpecialCharacters.java [hunk 10]
```diff
@@ -26,11 +27,19 @@
     private TestResults.Builder vehicleSubmittedData = TestResultsData.buildTestResultsSubmittedData();
```
Comment (cb-cs) [inline src/test/java/testresults/GetTestResultsVinSpecialCharacters.java:27]: These 2 are greyed out, are they still required?

File: src/test/java/testresults/PostTestResultsEmissionStandard.java [hunk 11]
```diff
@@ -25,11 +26,19 @@
     private TestResults.Builder vehicleSubmittedData = TestResultsData.buildTestResultsSubmittedData();
```
Comment (cb-cs) [inline src/test/java/testresults/PostTestResultsEmissionStandard.java:26]: Both greyed out here, are they still required?

File: src/test/java/testresults/PostTestResultsFuelType.java [hunk 12]
```diff
@@ -24,11 +25,19 @@
     private TestResults.Builder vehicleSubmittedData = TestResultsData.buildTestResultsSubmittedData();
```
Comment (cb-cs) [inline src/test/java/testresults/PostTestResultsFuelType.java:25]: Both greyed out here, are they still required?

File: src/test/java/testresults/PostTestResultsPozDefectsLvlCancelled.java [hunk 13]
```diff
@@ -34,12 +35,19 @@ private void validateSavedDataOld() {
         testResultsSteps.validateData((TestResultsGet) vehicleCancelledDataOld.build());
     }
 
+    private String test_results_cancelled_json;
```
Comment (cb-cs) [inline src/test/java/testresults/PostTestResultsPozDefectsLvlCancelled.java:38]: line 20 unused import

Also looks like validateSavedDataOld is no longer being used, is it still required?

File: src/main/java/data/GenericData.java [hunk 14]
```diff
@@ -456,4 +459,59 @@ public static JSONObject getRestrictions(String testCode) {
         }
         return actualRestrictions;
     }
+
+    public static String updateJson(TestResultsSteps testResultsSteps, String jsonFileName, String $) {
```
Comment (cb-cs) [inline src/main/java/data/GenericData.java:463]: $ is being passed through but not used.

File: src/main/java/data/GenericData.java (hunk 4 above)
Comment (wis-dvsa) [reply src/main/java/data/GenericData.java:461]: done

File: src/main/java/data/GenericData.java (hunk 14 above)
Comment (wis-dvsa) [reply src/main/java/data/GenericData.java:463]: done

File: src/test/java/testresults/GetTestResults.java (hunk 5 above)
Comment (wis-dvsa) [reply src/test/java/testresults/GetTestResults.java:45]: done

**Review by cb-cs — APPROVED** [review]

**Review by cb-cs — APPROVED** [review]

**Review by IuliaDVSA — APPROVED** [review]

**Pull request merged**